### PR TITLE
 gnrc_ipv6_ext: move ipv6_ext_rh (partly) to GNRC

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This moves all implementation specific parts of the `ipv6_ext_rh` module to `gnrc_ipv6_ext_rh`.

It also removes a module that isn't used after this change and doesn't contain any code (but isn't a pseudo-module for some reason).

### Testing procedure
Try to compile `gnrc_networking` with `USEMODULE = gnrc_rpl_srh`. It should still work.

Also (at least when run without DEBUG) `tests/gnrc_ipv6_ext` should still work without crashing (I will rework them once the integration process progressed).

### Issues/PRs references
Addresses parts of #10133 

PR dependencies

### CODE REVIEW

**Comment by miri64** [comment]
@bergzand could you maybe have a look (if doesn't intervene with the current release cycle)? @cgundogan is currently on vacation and this PR is giving me a really hard time, when I try to rebase #10238 on top of it and #10234 (which I do for testing).

**Comment by miri64** [comment]
Rebased and squashed current state, so tracking during the review is easier.

**Comment by bergzand** [comment]
The motivation for this change is to have these functions inside GNRC so that the structs/arguments can be kept GNRC-specific?

File: sys/include/net/gnrc/rpl/srh.h [hunk 1]
```diff
@@ -60,9 +60,9 @@ typedef struct __attribute__((packed)) {
  * @param[in,out] ipv6  The IPv6 header of the incoming packet.
  * @param[in] rh        A RPL source routing header.
  *
- * @return  EXT_RH_CODE_ERROR
- * @return  EXT_RH_CODE_FORWARD
- * @return  EXT_RH_CODE_OK
+ * @return  @ref GNRC_IPV6_EXT_RH_OK, on success
+ * @return  @ref GNRC_IPV6_EXT_RH_FORWARDED, when @p pkt *should be* forwarded
```
Comment (bergzand) [inline sys/include/net/gnrc/rpl/srh.h]: This contradicts the description of the enum above.

File: sys/include/net/ipv6/ext/rh.h [hunk 2]
```diff
@@ -9,13 +10,14 @@
...
  * @ingroup     net_ipv6_ext
- * @brief       Implementation of IPv6 routing header extension.
+ * @brief       Definitions for IPv6 routing header extension.
  * @{
  *
  * @file
  * @brief   Routing extension header definitions.
  *
- * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author  Cenk Gündoğan <cnkgndgn@gmail.com>
```
Comment (bergzand) [inline sys/include/net/ipv6/ext/rh.h]: Why this authorship addition and copyright addition above?

File: sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c [hunk 3]
```diff
@@ -65,30 +60,29 @@ static enum gnrc_ipv6_ext_demux_status _handle_rh(gnrc_pktsnip_t *current, gnrc_
...
-        case EXT_RH_CODE_OK:
+        case GNRC_IPV6_EXT_RH_OK:
             /* this should not happen since we checked seg_left early */
             gnrc_pktbuf_release(pkt);
-            return GNRC_IPV6_EXT_ERROR;
+            break;
     }
 
-    return GNRC_IPV6_EXT_OK;
+    return res;
```
Comment (bergzand) [inline sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c]: While technically speaking it should be impossible to reach this `return`, returning an unitialized variable that is never written here is not very pretty. Would it be okay to return `GNRC_IPV6_EXT_RH_ERROR` here as a sort of sane default?

**Comment by miri64** [comment]
Yes, and also that `ipv6_ext_rh_process()` should have never been outside of GNRC (the non-GNRC modules are just thought to be used for header parsing), as it straight jumps back into GNRC (for RPL SRH processing). Also there is a "GNRC-version" of this function which just adds another indirection that isn't necessary (see #10238).

File: sys/include/net/ipv6/ext/rh.h (hunk 2 above)
Comment (miri64) [reply sys/include/net/ipv6/ext/rh.h]: I did not write the original files, Cenk did (when he was a student at FU), so I changed it to reflect the actual copyright ;-).

Comment (bergzand) [reply sys/include/net/ipv6/ext/rh.h]: Ok :+1:

Comment (miri64) [reply sys/include/net/ipv6/ext/rh.h]: (have a look at this file's history to verify)

File: sys/include/net/gnrc/ipv6/ext/rh.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+enum {
+    /**
+     * @brief   The routing header was successfully processed and this node
+     *          is the destination
+     */
+    GNRC_IPV6_EXT_RH_OK = 0,
+    /**
+     * @brief   The routing header was successfully processed and the packet
+     *          was forwarded or should be forwarded by the central RH handler.
```
Comment (bergzand) [inline sys/include/net/gnrc/ipv6/ext/rh.h]: Something that "was forwarded" could have this node as destionation right? Does the `GNRC_IPV6_EXT_RH_OK` have priority over this result?

File: sys/include/net/gnrc/ipv6/ext/rh.h [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+    /**
+     * @brief   The routing header was successfully processed and this node
+     *          is the destination
+     */
+    GNRC_IPV6_EXT_RH_OK = 0,
```
Comment (bergzand) [inline sys/include/net/gnrc/ipv6/ext/rh.h]: Does it make sense to change the name here to something that indicates that the current node is the destination such as `GNRC_IPV6_EXT_RH_DEST` (Feel free to change to something that makes more sense to you, my lack of coffee prevents me from thinking of a sensible name)

Comment (miri64) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: `GNRC_IPV6_EXT_RH_AT_DST`?

Comment (bergzand) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: :+1:

Comment (miri64) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: (since this enum values are providing the most headache when rebasing, this will be fun to rename ;-D, but I think you are right regarding the naming)

Comment (bergzand) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: Sorry :(

Comment (miri64) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: Done

File: sys/include/net/gnrc/ipv6/ext/rh.h (hunk 4 above)
Comment (miri64) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: Can you rephrase this? I'm not sure I understand.

Comment (miri64) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: Ah I get it now

Comment (bergzand) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: Maybe I'm interpreting the "was forwarded" here not correct. To me it sounds as if a frame was forwarded by another node to this node. Where at that point both the conditions from this documentation and the option currently renamed to `GNRC_IPV6_EXT_RH_AT_DST` match the frame.

Comment (bergzand) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: Because it was both forwarded to this node and at the destination.

Comment (miri64) [reply sys/include/net/gnrc/ipv6/ext/rh.h]: Maybe the documentation update clarifies it better.

**Comment by miri64** [comment]
I changed the order of the enum, as it makes more sense when `= 0` is the error and `> 0` one of the two success cases.

**Comment by miri64** [comment]
I just noticed, that I did a terrible mistake with my fixup commits. May I squash?

**Comment by bergzand** [comment]
> May I squash?

Ack

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
I put the rename of `_OK` to `_AT_DST` to its own commit for clarity in case someone reviews the git history later ;-).

**Comment by miri64** [comment]
Adapted the unittests for the value name changes as well.

**Comment by miri64** [comment]
Release is feature frozen, so I'm merging this now.
